index.d.ts

```diff
@@ -0,0 +1,28 @@
+export default class Kareem {
+  static skipWrappedFunction(): SkipWrappedFunction;
+  static overwriteMiddlewareResult(): OverwriteMiddlewareResult;
+
+  pre(name: string | RegExp, fn: Function): this;
+  pre(name: string | RegExp, options: Record<string, any>, fn: Function, error?: any, unshift?: boolean): this;
+  post(name: string | RegExp, fn: Function): this;
+  post(name: string | RegExp, options: Record<string, any>, fn: Function, unshift?: boolean): this;
+
+  clone(): this;
```

Comment:
```suggestion
  clone(): Kareem;
```

the semantics suggest this returns a new separate instance, right?
(to my knowledge `this == Class name` (both are equal for typescript), but it should make suggestions (type hints) more obvious as `this` has the understanding of "existing instance" and "class name" as "separate (new) instance"; at least to me)

Reply:
Correct, will fix